client/post-editor/editor-term-selector/add-term.jsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+import TermFormDialog from 'blocks/term-form-dialog';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getPostTypeTaxonomy } from 'state/post-types/taxonomies/selectors';
+import { getTerms } from 'state/terms/selectors';
+
+class TermSelectorAddTerm extends Component {
+	static propTypes = {
+		labels: PropTypes.object,
+		terms: PropTypes.array
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			showDialog: false
+		};
+	}
+
+	openDialog = event => {
+		event.preventDefault();
+		this.setState( { showDialog: true } );
+	};
+
+	closeDialog = () => {
+		this.setState( { showDialog: false } );
+	}
+
+	render() {
+		const { labels, terms, ...props } = this.props;
+		const totalTerms = terms ? terms.length : 0;
+		const classes = classNames( 'editor-term-selector__add-term', { 'is-compact': totalTerms < 8 } );
+
+		return (
+			<div className={ classes }>
+				<Button borderless compact onClick={ this.openDialog }>
+					<Gridicon icon="folder" /> { labels.add_new_item }
+				</Button>
+				<TermFormDialog
+					showDialog={ this.state.showDialog }
+					onClose={ this.closeDialog }
+					{ ...props }
```

Comment:
It looks like `<TermFormDialog />` is already grabbing `terms` and `labels` on its own - should we just explicitly pass in `taxonomy` and `postType` here instead of all `...props`?